src/freenas/usr/local/lib/middlewared_truenas/plugins/failover.py

```diff
@@ -549,14 +549,14 @@ async def mismatch_disks(self):
         local_boot_disks = await self.middleware.call('boot.get_disks')
         remote_boot_disks = await self.middleware.call('failover.call_remote', 'boot.get_disks')
         local_disks = set(
-            v['ident']
-            for k, v in (await self.middleware.call('device.get_info', 'DISK')).items()
-            if k not in local_boot_disks
+            v['serial']
+            for k, v in (await self.middleware.call('device.retrieve_disks_data')).items()
```

Comment:
It just came to me, `retrieve_disks_data ` this does not exist in core device service, are we looking for freebsd support in failover code for master ?

Reply:
You're correct, I don't like calling `device.get_info 'DISK'` because it gets the rotationrate of every drive (by subprocessing out) which causes slowness. However, I don't see an easy way around it without making public API changes or doing an `if osc.IS_LINUX` check which is gross. I'll just go back to using the `get_info 'DISK'` API for now.